### PR TITLE
[vm] StarcoinVM refactor remove mut reference for some functions

### PR DESCRIPTION
**Pull request by welbon**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: vm/vm-runtime/src/data_cache.rs [hunk 1]
```diff
@@ -220,11 +219,11 @@ impl<S: StateView> TableResolver for RemoteStorageOwned<S> {
     }
 }
 
-impl<S: StateView> ConfigStorage for RemoteStorageOwned<S> {
-    fn fetch_config(&self, access_path: AccessPath) -> Option<Vec<u8>> {
-        self.as_move_resolver().fetch_config(access_path)
-    }
-}
+// impl<S: StateView> ConfigStorage for RemoteStorageOwned<S> {
```
Comment (nkysg) [inline vm/vm-runtime/src/data_cache.rs:222]: 这里是有冲突吧？

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 2]
```diff
@@ -902,9 +896,8 @@ impl StarcoinVM {
             MoveValue::U8(chain_id.id()),
             MoveValue::U64(parent_gas_used),
         ]);
-        let data_cache = remote_cache.as_move_resolver();
-        let mut session: SessionAdapter<_> =
-            self.move_vm.new_session(&data_cache, session_id).into();
+        //let data_cache = remote_cache.as_move_resolver();
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:899]: 这行不用就删了吧

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 3]
```diff
@@ -1079,13 +1070,13 @@ impl StarcoinVM {
...
-        state_view: &S,
+        storage: &S,
         transactions: Vec<Transaction>,
         block_gas_limit: Option<u64>,
     ) -> Result<Vec<(VMStatus, TransactionOutput)>> {
-        let mut data_cache = StateViewCache::new(state_view);
+        let mut data_cache = StateViewCache::new(storage);
         let mut result = vec![];
-        //TODO load config by config change event.
+        // load config by config change event
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:1079]: 这个TODO应该是没有实现吧？

File: vm/vm-runtime/src/starcoin_vm.rs [hunk 4]
```diff
@@ -1303,22 +1299,22 @@ impl StarcoinVM {
...
         &self,
         error_code: VMStatus,
         gas_meter: &mut StarcoinGasMeter,
         txn_data: &TransactionMetadata,
-        remote_cache: &StateViewCache<'_, S>,
+        storage: &S,
     ) -> (VMStatus, TransactionOutput) {
         gas_meter.set_metering(false);
-        let data_cache = remote_cache.as_move_resolver();
+        //let data_cache = remote_cache.as_move_resolver();
```
Comment (nkysg) [inline vm/vm-runtime/src/starcoin_vm.rs:1310]: 这行不用也删了吧

File: vm/vm-runtime/src/starcoin_vm.rs (hunk 3 above)
Comment (welbon) [reply vm/vm-runtime/src/starcoin_vm.rs:1079]: 加回去了

File: vm/vm-runtime/src/starcoin_vm.rs (hunk 2 above)
Comment (welbon) [reply vm/vm-runtime/src/starcoin_vm.rs:899]: 已处理

File: vm/vm-runtime/src/data_cache.rs (hunk 1 above)
Comment (welbon) [reply vm/vm-runtime/src/data_cache.rs:223]: 加上冲突说明的Todo了

**Review by nkysg — APPROVED** [review]

**Pull request merged**